diff --git a/cjs/react-dom-client.development.js b/cjs/react-dom-client.development.js
index 771ecad2064564b91b4ee66d5551b64f15500fad..d1bb4d8618775d00cad69b6211c1ed7c31cbcac6 100644
--- a/cjs/react-dom-client.development.js
+++ b/cjs/react-dom-client.development.js
@@ -3786,6 +3786,14 @@
           "_" !== key[0] &&
           addValueToProperties(key, object[key], properties, indent, prefix);
     }
+    function readTypeof(value) {
+      try {
+        return value.$$typeof;
+      } catch (error) {
+        return undefined;
+      }
+    }
+
     function addValueToProperties(
       propertyName,
       value,
@@ -3799,7 +3807,7 @@
             value = "null";
             break;
           } else {
-            if (value.$$typeof === REACT_ELEMENT_TYPE) {
+            if (readTypeof(value) === REACT_ELEMENT_TYPE) {
               var typeName = getComponentNameFromType(value.type) || "\u2026",
                 key = value.key;
               value = value.props;
@@ -3981,7 +3989,7 @@
                   "object" === typeof nextValue &&
                   null !== key &&
                   null !== nextValue &&
-                  key.$$typeof === nextValue.$$typeof
+                  readTypeof(key) === readTypeof(nextValue)
                 )
                   if (nextValue.$$typeof === REACT_ELEMENT_TYPE) {
                     if (
diff --git a/cjs/react-dom-profiling.development.js b/cjs/react-dom-profiling.development.js
index c8b583ce1709147000f5740c636aaf248d216e82..5c91f5d7b6b37e955c7f4c54a4cbcd40fc5d0bb6 100644
--- a/cjs/react-dom-profiling.development.js
+++ b/cjs/react-dom-profiling.development.js
@@ -3794,6 +3794,15 @@
           "_" !== key[0] &&
           addValueToProperties(key, object[key], properties, indent, prefix);
     }
+
+    function readTypeof(value) {
+      try {
+        return value.$$typeof;
+      } catch (error) {
+        return undefined;
+      }
+    }
+
     function addValueToProperties(
       propertyName,
       value,
@@ -3807,7 +3816,7 @@
             value = "null";
             break;
           } else {
-            if (value.$$typeof === REACT_ELEMENT_TYPE) {
+            if (readTypeof(value) === REACT_ELEMENT_TYPE) {
               var typeName = getComponentNameFromType(value.type) || "\u2026",
                 key = value.key;
               value = value.props;
@@ -3989,7 +3998,7 @@
                   "object" === typeof nextValue &&
                   null !== key &&
                   null !== nextValue &&
-                  key.$$typeof === nextValue.$$typeof
+                  readTypeof(key) === readTypeof(nextValue)
                 )
                   if (nextValue.$$typeof === REACT_ELEMENT_TYPE) {
                     if (
